japicmp/src/test/java/japicmp/cmp/InterfacesTest.java

```diff
@@ -187,6 +190,34 @@ public void testClassImplementsInterface() throws Exception {
 	}
 
 	@Test
+	public void testClassNoLongerImplementsInterface() throws Exception {
+		JarArchiveComparatorOptions jarArchiveComparatorOptions = new JarArchiveComparatorOptions();
+		jarArchiveComparatorOptions.setAccessModifier(AccessModifier.PRIVATE);
+		List<JApiClass> jApiClasses = ClassesHelper.compareClasses(jarArchiveComparatorOptions, new ClassesHelper.ClassesGenerator() {
+			@Override
+			public List<CtClass> createOldClasses(ClassPool classPool) throws Exception {
+				CtClass ctClassInterface = CtInterfaceBuilder.create().name("Interface").addToClassPool(classPool);
+				CtMethodBuilder.create().returnType(CtClass.voidType).publicAccess().abstractMethod().name("method").addToClass(ctClassInterface);
+				CtClass ctClass = CtClassBuilder.create().name("Test").implementsInterface(ctClassInterface).addToClassPool(classPool);
+				CtMethodBuilder.create().returnType(CtClass.voidType).publicAccess().name("method").addToClass(ctClass);
+				return Arrays.asList(ctClassInterface, ctClass);
+			}
+
+			@Override
+			public List<CtClass> createNewClasses(ClassPool classPool) throws Exception {
+				CtClass ctClassInterface = CtInterfaceBuilder.create().name("Interface").addToClassPool(classPool);
+				CtMethodBuilder.create().returnType(CtClass.voidType).publicAccess().abstractMethod().name("method").addToClass(ctClassInterface);
+				CtClass ctClass = CtClassBuilder.create().name("Test").addToClassPool(classPool);
+				return Arrays.asList(ctClassInterface, ctClass);
+			}
+		});
+		JApiClass jApiClass = getJApiClass(jApiClasses, "Test");
+		assertThat(jApiClass.isBinaryCompatible(), is(false));
+		assertThat(jApiClass.isSourceCompatible(), is(false));
+		assertThat(getJApiImplementedInterface(jApiClass.getInterfaces(), "Interface").getChangeStatus(), is(JApiChangeStatus.REMOVED));
```

Comment:
This sort of change is no longer tracked directly on the jApiClass as of https://github.com/siom79/japicmp/commit/cd40275849397f1c77e64f24ccc0d95020fb7933 which is why I had to update JApiClass.isBinaryCompatible and isSourceCompatible to check the compatibility changes in implemented interfaces.

However, I'll add assertions to all the new tests on the relevant jApiImplementedInterface and jApiSuperclass for the corresponding JApiCompatibilityChanges enum values in a follow-up PR.